crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractCustomResourceHandler.java

```diff
@@ -51,51 +52,55 @@ protected AbstractCustomResourceHandler(Resources resources, boolean parallel) {
-
-    visitTypeDef(def, visitors);
-
-    addDecorators(config, def, specReplicasPathDetector.getPath(),
-        statusReplicasPathDetector.getPath(), labelSelectorPathDetector.getPath());
-
-    Map<String, Property> additionalPrinterColumns = new HashMap<>(additionalPrinterColumnDetector.getProperties());
-    additionalPrinterColumns.forEach((path, property) -> {
-      Map<String, Object> parameters = property.getAnnotations().stream()
-          .filter(a -> a.getClassRef().getName().equals("PrinterColumn")).map(AnnotationRef::getParameters)
-          .findFirst().orElse(Collections.emptyMap());
-      String type = AbstractJsonSchema.getSchemaTypeFor(property.getTypeRef());
-      String column = (String) parameters.get("name");
-      if (Utils.isNullOrEmpty(column)) {
-        column = property.getName().toUpperCase();
+      List<Visitor<TypeDefBuilder>> visitors = new ArrayList<>();
+      if (config.specClassName().isPresent()) {
+        visitors.add(specReplicasPathDetector);
       }
-      String description = property.getComments().stream().filter(l -> !l.trim().startsWith("@"))
-          .collect(Collectors.joining(" ")).trim();
-      String format = (String) parameters.get("format");
-
-      resources.decorate(
-          getPrinterColumnDecorator(name, version, path, type, column, description, format));
-    });
+      if (config.statusClassName().isPresent()) {
+        visitors.add(statusReplicasPathDetector);
+      }
+      visitors.add(labelSelectorPathDetector);
+      visitors.add(additionalPrinterColumnDetector);
+      visitors.add(traversedClassesVisitor);
+
+      visitTypeDef(def, visitors);
+
+      addDecorators(config, def, specReplicasPathDetector.getPath(),
+          statusReplicasPathDetector.getPath(), labelSelectorPathDetector.getPath());
+
+      Map<String, Property> additionalPrinterColumns = new HashMap<>(additionalPrinterColumnDetector.getProperties());
+      additionalPrinterColumns.forEach((path, property) -> {
+        Map<String, Object> parameters = property.getAnnotations().stream()
+            .filter(a -> a.getClassRef().getName().equals("PrinterColumn")).map(AnnotationRef::getParameters)
+            .findFirst().orElse(Collections.emptyMap());
+        String type = AbstractJsonSchema.getSchemaTypeFor(property.getTypeRef());
+        String column = (String) parameters.get("name");
+        if (Utils.isNullOrEmpty(column)) {
+          column = property.getName().toUpperCase();
+        }
+        String description = property.getComments().stream().filter(l -> !l.trim().startsWith("@"))
+            .collect(Collectors.joining(" ")).trim();
+        String format = (String) parameters.get("format");
+
+        resources.decorate(
+            getPrinterColumnDecorator(name, version, path, type, column, description, format));
+      });
+    } catch (Exception e) {
+      throw new RuntimeException("Couldn't generate CRD for " + config.crdName(), e);
```

Comment:
I think that you can easily enough add a test 🙏

Reply:
It isn't trivial, actually. I actually cannot replicate the use case for which I added this for some reason (I suspect Lombok might play a role here 🤷🏼).